openmc/mesh.py

```diff
@@ -624,6 +628,333 @@ def to_xml_element(self):
 
         return element
 
+class CylindricalMesh(MeshBase):
+    """A 3D cylindrical mesh
+
+    Parameters
+    ----------
+    mesh_id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+    dimension : Iterable of int
+        The number of mesh cells in each direction.
+    n_dimension : int
+        Number of mesh dimensions (always 3 for a CylindricalMesh).
+    r_grid : Iterable of float
+        Mesh boundary points along the r-axis.
+    phi_grid : Iterable of float
+        Mesh boundary points along the phi-axis.
+    z_grid : Iterable of float
+        Mesh boundary points along the z-axis.
+    indices : Iterable of tuple
+        An iterable of mesh indices for each mesh element, e.g. [(1, 1, 1),
+        (2, 1, 1), ...]
+
+    """
+
+    def __init__(self, mesh_id=None, name=''):
+        super().__init__(mesh_id, name)
+
+        self._r_grid = None
+        self._phi_grid = [0, 360]
+        self._z_grid = None
+
+    @property
+    def dimension(self):
+        return (len(self.r_grid) - 1,
+                len(self.phi_grid) - 1,
+                len(self.z_grid) - 1)
+
+    @property
+    def n_dimension(self):
+        return 3
+
+    @property
+    def r_grid(self):
+        return self._r_grid
+
+    @property
+    def phi_grid(self):
+        return self._phi_grid
+
+    @property
+    def z_grid(self):
+        return self._z_grid
+
+    @property
+    def indices(self):
+        nr = len(self.r_grid) - 1
+        np = len(self.phi_grid) - 1
+        nz = len(self.z_grid) - 1
+        return ((r, p, z)
+                for z in range(1, nz + 1)
+                for p in range(1, np + 1)
+                for r in range(1, nr + 1))
+
+    @r_grid.setter
+    def r_grid(self, grid):
+        cv.check_type('mesh r_grid', grid, Iterable, Real)
+        self._r_grid = grid
+
+    @phi_grid.setter
+    def phi_grid(self, grid):
+        cv.check_type('mesh phi_grid', grid, Iterable, Real)
+        self._phi_grid = np.array(grid)
+
+    @z_grid.setter
+    def z_grid(self, grid):
+        cv.check_type('mesh z_grid', grid, Iterable, Real)
+        self._z_grid = np.array(grid)
+
+    def __repr__(self):
+        fmt = '{0: <16}{1}{2}\n'
+        string = super().__repr__()
+        string += fmt.format('\tDimensions', '=\t', self.n_dimension)
+        r_grid_str = str(self._r_grid) if self._r_grid is None else len(self._r_grid)
+        string += fmt.format('\tN R pnts:', '=\t', r_grid_str)
+        if self._r_grid is not None:
+            string += fmt.format('\tR Min:', '=\t', self._r_grid[0])
+            string += fmt.format('\tR Max:', '=\t', self._r_grid[-1])
+        t_grid_str = str(self._phi_grid) if self._phi_grid is None else len(self._phi_grid)
+        string += fmt.format('\tN Phi pnts:', '=\t', t_grid_str)
+        if self._phi_grid is not None:
```

Comment:
Unless I'm missing something, `self._phi_grid` should never be `None`.

Reply:
True for now. But I'm still thinking of some semantics, especially for `MeshSurfaceFilter`

At the moment phi = [0,360] means, there is no grid surface at phi=0. 
But then you can't define a Mesh where (only) at phi=0 the current is tallied.

This phi = None could be a different way to tell, that this axis is full 360°

So I would like to leave it there, as it does not really do any harm.